engine/worker/logger.go

```diff
@@ -146,15 +151,28 @@ func (w *currentWorker) grpcLogger(inputChan chan sdk.Log) error {
 		l, ok := <-inputChan
 		if ok {
 			log.Debug("LOG: %v", l.Val)
-			if err := stream.Send(&l); err != nil {
-				log.Error("grpcLogger> Error sending message : %s", err)
-				//Close all
-				stream.CloseSend()
-				w.grpc.conn.Close()
-				w.grpc.conn = nil
-				//Reinject log
-				inputChan <- l
-				return nil
+			if w.currentJob.wJob == nil {
+				if err := stream.Send(&l); err != nil {
```

Comment:
Maybe make just `if` to choose the right struct is more readable and less duplications